// client/src/contact/Contact.jsx
import React, { useState, useEffect, useRef } from 'react';
// 1. Remove EmailJS import
import './Contact.css';

const Contact = () => {
    const form = useRef();
    const [copied, setCopied] = useState(false);
    const [status, setStatus] = useState("");

    // --- Logic for Copying Email ---
    const handleCopyEmail = () => {
        const email = "ivangabrielromano8@gmail.com";
        navigator.clipboard.writeText(email);
        setCopied(true);
    };

    useEffect(() => {
        if (copied) {
            const timer = setTimeout(() => setCopied(false), 3000);
            return () => clearTimeout(timer);
        }
    }, [copied]);

    // --- NEW: Logic for Sending to your Backend ---
    const sendEmail = async (e) => {
        e.preventDefault();
        setStatus("Sending your request..."); // Provide immediate feedback

        // 2. Map form fields to a clean JSON object
        // Note: The keys here should match what your backend handler expects
        const formData = {
            name: e.target.user_name.value,
            email: e.target.user_email.value,
            company: e.target.company_name.value,
            headache: e.target.headache.value,
            message: e.target.message.value,
        };

        try {
            // 3. Make the API Call
            // If using the proxy we set up earlier, the path is just '/api/contact'
            const response = await fetch('/api/contact', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(formData),
            });

            const result = await response.json();

            if (response.ok) {
                setStatus("Success! Your solution is on the way.");
                e.target.reset(); // Clear fields
                setTimeout(() => setStatus(""), 6000);
            } else {
                // Handle server-side validation errors (like invalid email)
                setStatus(result.error || "Failed to send. Please try again.");
            }
        } catch (error) {
            // Handle network errors
            console.error("Submission Error:", error);
            setStatus("Server connection failed. Check your internet.");
        }
    };

    return (
        <section id="contact" className="relative w-full py-24 px-6 lg:px-20">

            {/* Toast Notification */}
            {copied && (
                <div className="fixed top-10 left-1/2 -translate-x-1/2 z-[100] bg-[#b88a44] text-white px-6 py-3 rounded-full shadow-2xl animate-bounce font-bold">
                    Email copied to clipboard! ✅
                </div>
            )}

            <div className="max-w-7xl mx-auto">
                <h2 className="text-4xl md:text-5xl font-bold text-white mb-16">Contact Us</h2>

                <div className="contact-outer-card p-8 lg:p-16 grid grid-cols-1 lg:grid-cols-2 gap-1 items-start">

                    {/* Left Side: Info */}
                    <div className="flex flex-col h-full justify-between py-4">
                        <div>
                            <h3 className="text-3xl md:text-4xl font-bold text-white mb-8 leading-tight">
                                Ready to <span className="text-[#b88a44]">Own</span> your <br /> digital future?
                            </h3>
                            <p className="text-gray-300 text-lg leading-relaxed mb-12 max-w-sm">
                                Let's identify your bottlenecks and architect a permanent solution.
                                No rent, no headaches—just results.
                            </p>
                        </div>

                        <div className="flex mb-12">
                            <a href="https://github.com/IGRP08" target="_blank" rel="noreferrer">
                                <img src="/assets/github-icon.png" alt="GitHub" className="w-30 h-30 social-icon" />
                            </a>
                            <div onClick={handleCopyEmail} className="cursor-pointer">
                                <img src="/assets/email-icon.png" alt="Email" className="w-30 h-30 social-icon" />
                            </div>
                            <a href="https://www.linkedin.com/in/ivan-romano-puente-870a78316/" target="_blank" rel="noreferrer">
                                <img src="/assets/linkedin-icon.png" alt="LinkedIn" className="w-30 h-30 social-icon" />
                            </a>
                        </div>

                        <p className="text-white font-bold text-xl leading-tight">
                            Infrastructure, Security, <br /> and Ownership <br /> guaranteed.
                        </p>
                    </div>

                    {/* Right Side: Form */}
                    <div className="contact-form-container p-8 md:p-10">
                        <form ref={form} onSubmit={sendEmail} className="space-y-6">
                            <div>
                                <label className="block text-white font-bold mb-2">Name</label>
                                <input type="text" name="user_name" placeholder="ex: Jacob Smith" className="contact-input" required />
                            </div>

                            <div>
                                <label className="block text-white font-bold mb-2">Email</label>
                                <input type="email" name="user_email" placeholder="ex: info@example.com" className="contact-input" required />
                            </div>

                            <div>
                                <label className="block text-white font-bold mb-2">Company Name</label>
                                <input type="text" name="company_name" placeholder="ex: Best Friend" className="contact-input" />
                            </div>

                            <div>
                                {/* The Label is back here */}
                                <label className="block text-white font-bold mb-2">Select Your Primary Headache</label>

                                <div className="relative">
                                    <select
                                        name="headache"
                                        className="contact-input appearance-none w-full pr-10"
                                    >
                                        <option value="Technical Debt">Web Application</option>
                                        <option value="Legacy Systems">Business Email Solution</option>
                                        <option value="Slow Website">Domain Name (URL)</option>
                                        <option value="Poor SEO">Google Search Positioning</option>
                                    </select>

                                    {/* Custom Gold Arrow Icon */}
                                    <div className="pointer-events-none absolute inset-y-0 right-0 flex items-center px-3 text-[#b88a44]">
                                        <svg className="h-5 w-5 fill-current" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20">
                                            <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
                                        </svg>
                                    </div>
                                </div>
                            </div>

                            <div>
                                <label className="block text-white font-bold mb-2">Describe your project</label>
                                <textarea name="message" rows="4" className="contact-input resize-none" required></textarea>
                            </div>

                            <div className="flex flex-col items-end gap-4">
                                <button
                                    type="submit"
                                    className="btn-gold disabled:opacity-50"
                                    disabled={status === "Sending your request..."}
                                >
                                    Build Your Solution
                                </button>
                                {status && <p className="text-[#b88a44] font-bold italic">{status}</p>}
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        </section>
    );
};

export default Contact;
